packages/notebook/src/model.ts

```diff
@@ -428,6 +429,27 @@ close the notebook without saving it.`,
         this.triggerStateChange(value);
       });
     }
+
+    if (changes.metadataChange) {
+      const metadata = changes.metadataChange.newValue as JSONObject;
+      this._modelDBMutex(() => {
+        Object.entries(metadata).forEach(([key, value]) => {
+          this.metadata.set(key, value);
+        });
+      });
+    }
+  }
+
+  private _onMetadataChanged(
+    metadata: IObservableJSON,
+    change: IObservableMap.IChangedArgs<ReadonlyPartialJSONValue | undefined>
+  ): void {
+    if (change.key !== 'kernelspec' && change.key !== 'language_info') {
```

Comment:
I would set that as constant at the module level:

```suggestion
    if ( !UNSHARED_KEYS.includes(change.key) ) {
```

Reply:
Hey @fcollonval, thanks for the suggestion. I applied the changes on my last commit.